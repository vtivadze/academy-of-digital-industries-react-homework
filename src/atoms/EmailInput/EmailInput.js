import { useState, useEffect } from "react";
import { getFormInputClassName } from "../../helpers";

export const EmailInput = () => {
  const [email, setEmail] = useState("");
  const [emailError, setEmailError] = useState(false);

  useEffect(() => {
    const emailPattern = /^[a-z]+[0-9]*[a-z]*@[a-z]{2,}\.[a-z]{2,3}$/im;

    if (email && !email.match(emailPattern)) {
      setEmailError(true);
    } else if (email && email.match(emailPattern)) {
      setEmailError(false);
    }
  }, [email]);

  return (
    <div className="control has-icons-left has-icons-right">
      <input
        className={getFormInputClassName(emailError, email)}
        name="email"
        value={email}
        type="email"
        placeholder="Email"
        autoComplete="email"
        required
        onChange={({target}) => setEmail(target.value) }
      />

      <span className="icon is-small is-left">
        <i className="fas fa-envelope"></i>
      </span>
      {!emailError && email && <span className="icon is-small is-right">
        <i className="fas fa-check"></i>
      </span>}
      {emailError && email && 
        <>
          <span className="icon is-small is-right">
            <i className="fas fa-exclamation-triangle"></i>
          </span>
          <p className="help is-danger">This email is invalid</p>
        </>
      }
    </div>
  );
};