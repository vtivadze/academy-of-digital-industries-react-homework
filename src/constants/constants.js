export const AUTH_TOKEN = "auth:user_token";