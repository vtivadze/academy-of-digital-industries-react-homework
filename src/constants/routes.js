export const HOME_PATH = "/";
export const ABOUT_PATH = "/about";
export const SERVICES_PATH = "/services";
export const STAFF_PATH = "/staff";
export const LOGIN_PATH = "/login";
export const SIGNUP_PATH = "/signup";
export const CHECK_EMAIL_PATH = "/check-email";
export const SET_NEW_PASSWORD_PATH = "/set-new-password";
export const COUNTERS_PATH = "/counters";
export const USERS_PATH = "/users";
export const PROFILE_PATH = "/profile";
export const ACCESS_IS_DENIED_PATH = "/access-is-denied";