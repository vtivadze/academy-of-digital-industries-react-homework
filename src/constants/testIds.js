export const TEST_IDS = {
  emailInput: {
    inputElement: 'emailInput',
    hasError: 'email.hasError',
    success: 'email.success',
  },
};