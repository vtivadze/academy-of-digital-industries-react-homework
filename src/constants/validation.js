export const EMAIL_MIN_LENGTH = 8;
export const EMAIL_MAX_LENGTH = 50;
export const EMAIL_PATTERN = /^[a-z]+[0-9]*[a-z]*@[a-z]{2,}\.[a-z]{2,3}$/im;

export const NAME_MIN_LENGTH = 2;
export const NAME_MAX_LENGTH = 51;
export const NAME_PATTERN = /^[a-z]{2,20}\s+[a-z]{2,30}$/im;

export const USERNAME_MIN_LENGTH = 2;
export const USERNAME_MAX_LENGTH = 20;
export const USERNAME_PATTERN = /^[a-z0-9]{2,20}$/im;

export const PASSWORD_MIN_LENGTH = 10;
export const PASSWORD_MAX_LENGTH = 20;
