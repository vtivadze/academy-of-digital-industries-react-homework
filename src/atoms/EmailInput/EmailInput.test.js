import { render } from '@testing-library/react';
import { TEST_IDS } from '../../constants/testIds';
import { EmailInput } from './EmailInput';

describe('EmailInput', () => {
  it('should render correctly with correct props', () => {
    const mockedName = 'mocked-name';
    const Component = render(<EmailInput inputName={mockedName} />);

    expect(
      Component.getByTestId(TEST_IDS.emailInput.inputElement).getAttribute(
        'name'
      )
    ).toBe(mockedName);
  });
});

// describe('EmailInput snapshots', () => {
//   describe('snap 1', () => {
//     const mockedName = 'mocked-name';
//     const Component = render(<EmailInput inputName={mockedName} />);

//     expect(Component).toMatchInlineSnapshot();
//   });
// });
