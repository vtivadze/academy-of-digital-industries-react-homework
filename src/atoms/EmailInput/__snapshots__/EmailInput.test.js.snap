// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput shapshot snap 1 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="control has-icons-left has-icons-right"
      >
        <input
          autocomplete="email"
          class="input"
          data-testid="emailInput"
          name="mocked-name"
          placeholder="Email"
          required=""
          type="email"
          value=""
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fas fa-envelope"
          />
        </span>
        
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="control has-icons-left has-icons-right"
    >
      <input
        autocomplete="email"
        class="input"
        data-testid="emailInput"
        name="mocked-name"
        placeholder="Email"
        required=""
        type="email"
        value=""
      />
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-envelope"
        />
      </span>
      
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`EmailInput shapshot snap 2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="control has-icons-left has-icons-right"
      >
        <input
          autocomplete="email"
          class="input is-danger"
          data-testid="emailInput"
          name="mocked-name"
          placeholder="Email"
          required=""
          type="email"
          value="wrong"
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fas fa-envelope"
          />
        </span>
        <span
          class="icon is-small is-right"
          data-testid="email.hasError"
        >
          <i
            class="fas fa-exclamation-triangle"
          />
        </span>
        <p
          class="help is-danger"
        >
          This email is invalid
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="control has-icons-left has-icons-right"
    >
      <input
        autocomplete="email"
        class="input is-danger"
        data-testid="emailInput"
        name="mocked-name"
        placeholder="Email"
        required=""
        type="email"
        value="wrong"
      />
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-envelope"
        />
      </span>
      <span
        class="icon is-small is-right"
        data-testid="email.hasError"
      >
        <i
          class="fas fa-exclamation-triangle"
        />
      </span>
      <p
        class="help is-danger"
      >
        This email is invalid
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`EmailInput shapshot snap 3 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="control has-icons-left has-icons-right"
      >
        <input
          autocomplete="email"
          class="input is-success"
          data-testid="emailInput"
          name="mocked-name"
          placeholder="Email"
          required=""
          type="email"
          value="test@test.com"
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fas fa-envelope"
          />
        </span>
        <span
          class="icon is-small is-right"
          data-testid="email.success"
        >
          <i
            class="fas fa-check"
          />
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="control has-icons-left has-icons-right"
    >
      <input
        autocomplete="email"
        class="input is-success"
        data-testid="emailInput"
        name="mocked-name"
        placeholder="Email"
        required=""
        type="email"
        value="test@test.com"
      />
      <span
        class="icon is-small is-left"
      >
        <i
          class="fas fa-envelope"
        />
      </span>
      <span
        class="icon is-small is-right"
        data-testid="email.success"
      >
        <i
          class="fas fa-check"
        />
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
